### ISSUE TITLE
Sending images from bytes.

### ISSUE THREAD

**Issue by Highstaker** [python-telegram-bot/python-telegram-bot#119]
Sending images from existing files works fine. But if I want to send it from bytes stored in variable, there is no way to do it. Bytes are not JSON-serializable in Python3.

I tried 
`>>> f=io.BufferedReader(BytesIO(b))`
`>>> bot.sendPhoto(chat_id=chat_id,photo=f)`
where `b` contains image file contents as bytes (downloaded from internet, for example). decoding it doesn't work.
But it says 
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.4/dist-packages/telegram/bot.py", line 126, in decorator
    result = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/telegram/bot.py", line 158, in decorator
    result = request.post(url, data)
  File "/usr/local/lib/python3.4/dist-packages/telegram/utils/request.py", line 95, in post
    data = InputFile(data)
  File "/usr/local/lib/python3.4/dist-packages/telegram/inputfile.py", line 76, in __init__
    self.filename = os.path.basename(self.input_file.name)
AttributeError: '_io.BytesIO' object has no attribute 'name'
```
Guess the code requires a filename.

Can the sending from bytes function be made?

**Comment by leandrotoledo** [comment]
Thanks for reporting in!
And sorry it took so long, the fix has been submitted on PR #199 for review.

**Comment by leandrotoledo** [comment]
#199 has been merged!
Closing this one.

**Comment by apetrin** [comment]
Hi there! I afraid issue still remains. Here is what I'm 
```
In [9]: telegram.__version__ 
Out[9]: '3.4'
In [10]: bot = telegram.Bot('<token>')
In [11]: chat_id = 116525440
In [12]: kitty = open('fail.jpg','rb')
In [13]: f=io.BufferedReader(io.BytesIO(kitty.read()))
In [14]: bot.sendPhoto(chat_id, f)
```

And it throws
```
TypeError: <_io.BufferedReader> is not JSON serializable
```

Can you please check that you can replicate this error?

**Comment by jh0ker** [comment]
It's working for me in Python 3.4, but I get the same Error as you in Python 2.7

**Comment by leandrotoledo** [comment]
Indeed this seems to be related to Python 2 only. Mostly because fileinput.py#201 ``isinstance(file_content, file)``. A quick solution would be checking for ``isinstance(file_content, BufferedReader)``. Altought, I'd rather have fileinput.py refactored and remove line fileinput.py#26 ``from io import BufferedReader as file``.

I'll work in something in that way.

**Comment by aidarbiktimirov** [comment]
What about `hasattr(file_content, 'read')`?
http://stackoverflow.com/a/1661354

**Comment by leandrotoledo** [comment]
Kudos on @aidarbiktimirov!
252abb1 on PR #247 should fix the problem and get rid of `from io import BufferedReader as file`, tests have been added as well.

This should be included on v4.0 final release.

**Comment by jh0ker** [comment]
Closing
